api/objects/validate

```diff
@@ -79,18 +79,29 @@ case "update-service":
     }
 
     if($data['Ports']) {
+        $portValidator = $v->createValidator(Validate::PORTNUMBER);
         if (strpos($data['Ports'][0], "-") !== false ) { # port range
-            $tmp = explode("-",$data['Ports'][0]);
-            # two non-zero integers, right number must be greater then left one
-            if ( !isset($tmp[0], $tmp[1]) || !(int)$tmp[0] || !(int)$tmp[1] || (int)$tmp[0] >= (int)$tmp[1] ) {
-                $v->addValidationError('Ports', 'not_valid_port_range');
+            # Verify that we use either a comma separated list OR a range of ports
+            if (isset ($data['Ports'][1])) {
+                $v->addValidationError('Ports', 'must_be_port_list_or_a_port_range');
+            } else {
+                $tmp = explode("-",$data['Ports'][0]);
```

Comment:
Limit the array length to 2: more splits is an error.
```suggestion
                $tmp = explode("-", $data['Ports'][0], 2);
```

With a length limit it could be nice to use a variables list:

    list($lower_port, $upper_port) = explode("-", $data['Ports'][0], 2);

Reply:
it should be good